lib/Cake/View/Helper/FormHelper.php

```diff
@@ -1638,7 +1645,12 @@ public function radio($fieldName, $options = array(), $attributes = array()) {
 			$between = '';
 		}
 		if ($legend) {
-			$out = $this->Html->useTag('fieldset', '', $this->Html->useTag('legend', $legend) . $between . $out);
+			if (is_string($fieldset)) {
+				$fieldsetClass = sprintf(' class="%s"', $fieldset);
```

Comment:
Because as an array `useTag()` will call `_parseAttributes` which will then escape the value.. I think I understand now. I'll send another update once I made the changes and wrote the test